### ISSUE TITLE
Create DOMO Dashboard View of Open Issues by Product

### ISSUE THREAD

**Issue by chaseakirby** [department-of-veterans-affairs/va.gov-team#34739]
## User Story
As a Governance Team member, I want to see the number of issues by product so that I understand known issues.

As an OCTO-DE lead, I want to see number of issues by product so that I can fully understand the scope of issues across products brought through Collaboration Cycle.

- Intended Audience: Governance Team and OCTO-DE leads
- Desired Outcome: Shared understanding of product health across VA.gov
- Due Date: TBD

## Description
We currently have a ["Total Issues by Team" DOMO board](https://va-gov.domo.com/page/1340527018?userId=465116418).
With our move to Product focus, we will need new boards and cards providing data on the product labels added to staging reviews in ticket [#37783](https://app.zenhub.com/workspaces/platform-governance-team-5f85b91c14d8df0018fac414/issues/department-of-veterans-affairs/va.gov-team/37783).

Additional background for [Collaboration Cycle - Improve tracking and status visibility of VA.gov products](https://app.zenhub.com/workspaces/platform-governance-team-5f85b91c14d8df0018fac414/issues/department-of-veterans-affairs/va.gov-team/34697) initiative <br />

## New Artifacts 
New DOMO Board for Product Views
Total Issues by Product Card
Total Issues by Status

## Tasks
- [x] Create DOMO board for Product Views
- [x] Add Total Issues by Product Card
- [ ] Change name to "Total Open Issues by Product"
- [ ] Change pie chart to bar graph
- [ ] Map labels to an official product name
- [ ] Request support from Insight & Analytics
  - [ ] Ticket #42627 opened for support
  - [x] Meeting with Analytics & Insights on 6/29. [Link to recording](https://dsva.slack.com/archives/CKCRMLGAC/p1656526968126789)
- [x] Add Total Issues by Status
- [x] Document Queries used and other information for updating/maintenance
- [ ] Delete/Archive the "Total Issues by Team" card

## Acceptance Criteria
- [ ] Total Issues by Status card should be editable and able to be maintained by Sarah
- [ ] Labels should be mapped to an official product name
- [ ] Dashboard should display all open staging review issues by product

---

## Related Governance Team Resources
_For additional information and guidance to complete tasks, please see links below._
[Documentation Update Guidance for GitHub templates](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2047868956/Guidance+for+updating+GitHub+Issue+templates) <br />
[Documentation Update Guidance for Platform Website pages](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2089091115/Guidance+for+updating+Platform+Website+pages) <br />
[Documentation Update Guidance for Confluence pages](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2089222173/Guidance+for+updating+Confluence+pages) <br />
[Guidelines for Governance Team Confluence Space](https://vfs.atlassian.net/wiki/spaces/AP/pages/1127219219/Guidelines+for+Team+spaces) <br />
[Guidelines for Platform Products and Services Confluence Space](https://vfs.atlassian.net/wiki/spaces/AP/pages/1321140361/Guidelines+for+Platform+Products+and+Services+space) <br />
[Description of Platform Website spaces and content](https://vfs.atlassian.net/wiki/spaces/AP/pages/1241579796/Organization) <br />

---
## How to configure this issue
- [ ] **Ensure that story is attached to appropriate Iniative (if not tech debt)** (what body of work is this a part of?)
- [ ] Ensure that any additional labels are added ('tech-debt', 'collaboration-cycle')
- [ ] Ensure that story is in the correct pipeline (Backlog, Planned, In Progress, Review/QA, Closed)
- [ ] Ensure that assignees are correct so you receive all updates/comments on the ticket

**Comment by SarahKay8** [comment]
Currently waiting on BigQuery/Domo query permissions. Foundational cards have been added to Dashboard.. just need to apply a few edits to the SQL Ryan sent over. Almost done here. @shiragoodman @chaseakirby

**Comment by shiragoodman** [comment]
I scheduled a touchbase with @Saraharaba8 @chaseakirby and I for Thursday 4/21 at 2:30pm ET.

**Comment by SarahKay8** [comment]
Currently making modifications to SQL template presented by Ryan Johnson.

**Comment by SarahKay8** [comment]
Modifications to SQL have been implemented. Code has been sent to Chase via slack for review.

**Comment by SarahKay8** [comment]
SQL verified by @chaseakirby 
05/02/2022

Currently setting up access to Github_issues dataset with Brian Martin.

He is currently in the process of creating a JSON key so I have full access

**Comment by SarahKay8** [comment]
Received JSON Key from Brian Martin

**Comment by SarahKay8** [comment]
Tried to run SQL via DOMO and experiencing error message below. Awaiting next step procedures from Ryan Johnson. Tried to run original template he provided and still getting the same issue.. 

Error message for reference: ![Image 5-11-22 at 4.40 PM.jpg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/9dc0398c-5779-4ae7-a323-93f5b134156b)

**Comment by SarahKay8** [comment]
Github Product Label data has successfully been extracted from query. Visualization has been edited. Ready for review. @shiragoodman

**Comment by SarahKay8** [comment]
SQL used to retrieve data from Github database: `with selected as (
SELECT number, state, labels
FROM `vsp-analytics-and-insights.vsp_testing_tools.github_issues`,
    UNNEST(labels) as collab_cycle_feedback,
    UNNEST(labels) as practice_area,
    UNNEST(labels) as experience_standard,
    UNNEST(labels) as touchpoint
WHERE collab_cycle_feedback = 'collab-cycle-feedback'
    and practice_area in ('QA', 'IA', 'Content', 'Design', '508/Accessibility')
    and experience_standard like 'CCIssue%'
    and touchpoint in ('Staging','Midpoint')
),
oc_counts as (select 
countif(state = 'closed' AND unnestLabels = 'Caregivers') as Caregivers_closed,
countif('launch-blocking' IN unnest(labels) and 'Caregivers' IN unnest(labels) and 'open' IN (state)) as Caregivers_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'Caregivers' IN unnest(labels) and 'open' IN (state)) as Caregivers_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'gi_comparison_tool') as gi_comparison_tool_closed,
countif('launch-blocking' IN unnest(labels) and 'gi_comparison_tool' IN unnest(labels) and 'open' IN (state)) as gi_comparison_tool_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'gi_comparison_tool' IN unnest(labels) and 'open' IN (state)) as gi_comparison_tool_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'notifications_preferences') as notifications_preferences_closed,
countif('launch-blocking' IN unnest(labels) and 'notifications_preferences' IN unnest(labels) and 'open' IN (state)) as notifications_preferences_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'notifications_preferences' IN unnest(labels) and 'open' IN (state)) as notifications_preferences_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'HLR') as HLR_closed,
countif('launch-blocking' IN unnest(labels) and 'HLR' IN unnest(labels) and 'open' IN (state)) as HLR_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'HLR' IN unnest(labels) and 'open' IN (state)) as HLR_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'MCP') as MCP_closed,
countif('launch-blocking' IN unnest(labels) and 'MCP' IN unnest(labels) and 'open' IN (state)) as MCP_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'MCP' IN unnest(labels) and 'open' IN (state)) as MCP_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'dt_find_va_form') as dt_find_va_form_closed,
countif('launch-blocking' IN unnest(labels) and 'dt_find_va_form' IN unnest(labels) and 'open' IN (state)) as dt_find_va_form_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'dt_find_va_form' IN unnest(labels) and 'open' IN (state)) as dt_find_va_form_open_not_blocked,
countif(state = 'closed' AND unnestLabels = '686') as six_eight_six_closed,
countif('launch-blocking' IN unnest(labels) and '686' IN unnest(labels) and 'open' IN (state)) as six_eight_six_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and '686' IN unnest(labels) and 'open' IN (state)) as six_eight_six_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'sign_in_page') as sign_in_page_closed,
countif('launch-blocking' IN unnest(labels) and 'sign_in_page' IN unnest(labels) and 'open' IN (state)) as sign_in_page_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'sign_in_page' IN unnest(labels) and 'open' IN (state)) as sign_in_page_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'HCE_Checkin') as HCE_Checkin_closed,
countif('launch-blocking' IN unnest(labels) and 'HCE_Checkin' IN unnest(labels) and 'open' IN (state)) as HCE_Checkin_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'HCE_Checkin' IN unnest(labels) and 'open' IN (state)) as HCE_Checkin_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'HRT') as HRT_closed,
countif('launch-blocking' IN unnest(labels) and 'HRT' IN unnest(labels) and 'open' IN (state)) as HRT_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'HRT' IN unnest(labels) and 'open' IN (state)) as HRT_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'outreach and events') as  outreach and events_closed,
countif('launch-blocking' IN unnest(labels) and 'outreach and events' IN unnest(labels) and 'open' IN (state)) as  outreach and events_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'outreach and events' IN unnest(labels) and 'open' IN (state)) as  outreach and events_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'typeahead') as  typeahead_closed,
countif('launch-blocking' IN unnest(labels) and 'typeahead' IN unnest(labels) and 'open' IN (state)) as  typeahead_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'typeahead' IN unnest(labels) and 'open' IN (state)) as  typeahead_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'vanotify') as  vanotify_closed,
countif('launch-blocking' IN unnest(labels) and 'vanotify' IN unnest(labels) and 'open' IN (state)) as  vanotify_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'vanotify' IN unnest(labels) and 'open' IN (state)) as  vanotify_open_not_blocked,
countif(state = 'closed' AND unnestLabels = 'Virtual_Agenty') as  Virtual_Agent_closed,
countif('launch-blocking' IN unnest(labels) and 'Virtual_Agent' IN unnest(labels) and 'open' IN (state)) as  Virtual_Agent_open_blocked,
countif('launch-blocking' NOT IN unnest(labels) and 'Virtual_Agent' IN unnest(labels) and 'open' IN (state)) as  Virtual_Agent_open_not_blocked,
from selected,
UNNEST(labels) AS unnestLabels
)
 select 'Caregivers' as name, Caregivers_open_blocked as open_blocked, Caregivers_open_not_blocked as open_not_blocked, Caregivers_closed as closed from oc_counts
 union all select 'gi_comparison_tool' as name, gi_comparison_tool_open_blocked as open_blocked, gi_comparison_tool_open_blocked as open_not_blocked, gi_comparison_tool_closed as closed from oc_counts
 union all select 'notifications_preferences' as name, notifications_preferences_open_blocked as open_blocked, notifications_preferences_open_blocked as open_not_blocked, notifications_preferences_closed as closed from oc_counts
 union all select 'HLR' as name, HLR_open_blocked as open_blocked, HLR_open_blocked as open_not_blocked, HLR_closed as closed from oc_counts
 union all select 'MCP' as name, MCP_open_blocked as open_blocked, MCP_open_not_blocked as open_not_blocked, MCP_closed as closed from oc_counts
 union all select 'Dt_find_va_form)' as name, Dt_find_va_form_open_blocked as open_blocked, Dt_find_va_form_open_not_blocked as open_not_blocked, Dt_find_va_form_closed as closed from oc_counts
 union all select '686' as name, 686_open_blocked as open_blocked, six_eight_six_open_not_blocked as open_not_blocked, six_eight_six_closed as closed from oc_counts
 union all select 'Sign_in_page' as name, Sign_in_page_open_blocked as open_blocked, Sign_in_page_not_blocked as open_not_blocked, Sign_in_page_closed as closed from oc_counts
 union all select 'HCE_Checkin' as name,HCE_Checkin_open_blocked as open_blocked, HCE_Checkin_open_not_blocked as open_not_blocked, HCE_CHeckin_closed as closed from oc_counts
 union all select 'HRT' as name, HRT_open_blocked as open_blocked, HRT_open_not_blocked as open_not_blocked, HRT_closed as closed from oc_counts
 union all select 'Outreach and Events' as name, Outreach and Events_open_blocked as open_blocked, Outreach and Events_open_not_blocked as open_not_blocked, Outreach and Events_closed as closed from oc_counts
 union all select 'typeahead' as name, typeahead_open_blocked as open_blocked, typeahead_open_not_blocked as open_not_blocked, typeahead_closed as closed from oc_counts
 union all select 'vanotify' as name, vanotify_open_blocked as open_blocked, vanotify_open_not_blocked as open_not_blocked, vanotify_closed as closed from oc_counts
 union all select 'virtual_agent' as name,  virtual_agent_open_blocked as open_blocked, virtual_agent_open_not_blocked as open_not_blocked,  virtual_agent_closed as closed from oc_counts`

**Comment by chaseakirby** [comment]
@SarahKay8 Hey Sarah, I don't see the Dashboard to view by Products. Can you comment a link to the board/card?

**Comment by SarahKay8** [comment]
HI @chaseakirby.Does this url populate the card? https://va-gov.domo.com/analyzer?cardid=1350189801

**Comment by chaseakirby** [comment]
I get a "this card hasn't been shared with you."
I think that we should put the card into a Dashboard, it appears that DOMO only displays Dashboards on the left navigation and cards may need a specific url to access. 

![image](https://user-images.githubusercontent.com/94856753/169097708-72e79c74-20f7-4bbd-9eff-d3deaa44aec2.png)

**Comment by SarahKay8** [comment]
> I get a "this card hasn't been shared with you." I think that we should put the card into a Dashboard, it appears that DOMO only displays Dashboards on the left navigation and cards may need a specific url to access.
> 
> ![image](https://user-images.githubusercontent.com/94856753/169097708-72e79c74-20f7-4bbd-9eff-d3deaa44aec2.png)

Hi @chaseakirby try this url https://va-gov.domo.com/page/366499128. I added the card to the dashboard I created

**Comment by chaseakirby** [comment]
@Saraharaba8 That links tells me that the page is private.
Is there a setting to make it public? I don't see a need to keep this private as we currently screenshot the "open issues by team" card for our weekly Team of Teams meeting.

![image](https://user-images.githubusercontent.com/94856753/169351134-ab361430-4edd-49cb-8ad8-398556782a7a.png)

**Comment by SarahKay8** [comment]
@chaseakirby there's a padlock icon and its currently unlocked. So I'm assuming it's public. I shared the dashboard with you. Check your spam if you don't see it in primary. Thanks Chase

**Comment by chaseakirby** [comment]
@SarahKay8 Thanks! I can see it now. I only see data for two of the labels (caregivers & 686) so I think we may need to see if the other ones are being found by the DataSet or being filtered out somewhere. 
I also see under Details that it shows only us as having access to the card.
![image.png](https://images.zenhubusercontent.com/619d1b3396a3389c5e2dd17f/1910f577-7207-4d05-8219-e3bdd8149d66)

**Comment by SarahKay8** [comment]
@chaseakirby gotcha. Should we set something up?

**Comment by chaseakirby** [comment]
@SarahKay8 Yeah I think we can go through some and do some more tests on it. I'll schedule a meeting for us.

**Comment by SarahKay8** [comment]
@chaseakirby sounds good! See you then

**Comment by SarahKay8** [comment]
@chaseakirby All label data is present. Ticket ready for review

**Comment by chaseakirby** [comment]
@SarahKay8 Thanks Sarah! Is this the same link, https://va-gov.domo.com/page/366499128, as before or did you have to create a new card with a new link?
The existing link shows only the 686 and Caregivers data still.

**Comment by SarahKay8** [comment]
@chaseakirby Here you are! https://va-gov.domo.com/page/366499128?userId=523054097

**Comment by chaseakirby** [comment]
@SarahKay8 Thanks!! So that one definitely has more data (which is great!) but the numbers don't seem to line up just yet. Our existing dashboards use the CC-Dashboard label to populate the cards so I wonder if that is doing something since this card is on that dashboard. #34244 may be a good candidate to look at since it has 6 open Staging Issue tickets but those don't seem to be in the DOMO card. 
Also, here is a Query of the [open CC-Dashboard tickets](https://github.com/department-of-veterans-affairs/va.gov-team/issues?q=is%3Aopen+is%3Aissue+label%3ACC-Dashboard), DOMO shows 136 and this shows 81 open so I think we may need to do some more spot checking.

**Comment by SarahKay8** [comment]
Ok makes sense! Should we plan another troubleshooting sync? Or should I reach out to analytics? Jonathan E. assisted me with the most recent SQL so I can forward this comment over to him

**Comment by chaseakirby** [comment]
@SarahKay8 I added a follow up comment as well. I think that we can tag up and make sure that we are on the same page and see the same things and then continue working with Analytics.

**Comment by SarahKay8** [comment]
Sounds good! @chaseakirby

**Comment by shiragoodman** [comment]
@Saraharaba8 can you grant me access to view the dashboard?

**Comment by SarahKay8** [comment]
Sure one sec @shiragoodman

**Comment by SarahKay8** [comment]
> @Saraharaba8 can you grant me access to view the dashboard?

Sent

**Comment by shiragoodman** [comment]
@SarahKay8 just realized i tagged your old Github account. Whoops! My bad.

I'd love to set up a meeting to discuss/review where we're currently at and what is needed to close this out. It may also be a good opportunity to identify what enhancements we can make next to the dashboard. 

I'll be sending out a meeting invite soon.

**Comment by SarahKay8** [comment]
> @SarahKay8 just realized i tagged your old Github account. Whoops! My bad.
> 
> 
> 
> I'd love to set up a meeting to discuss/review where we're currently at and what is needed to close this out. It may also be a good opportunity to identify what enhancements we can make next to the dashboard. 
> 
> 
> 
> I'll be sending out a meeting invite soon.

@shiragoodman Sounds good!

**Comment by shiragoodman** [comment]
- Not all of the tickets are showing up/being pulled in. for example, HLR has 4 open tickets that are not being pulled in.
- Jonathan Epstein helped with SQL quickly.
- Total Issue by Team card can be deleted/archived.
 - Github Product Labels name change to "Total Open Issues by Product"
 - Pie chart should change to bar graph
- map our labels to an official product name
- Total Issues by Status card should be editable and able to be maintained by Sarah

**Comment by shiragoodman** [comment]
Closing this issue and tracking in ticket #42627
